Fixed firebear category import logic + disabled inserting new categories on import. Ticket number: IPET-2372

diff --git a/Model/Import/Category.php b/Model/Import/Category.php
index 5b4f9ed..13c2c4d 100644
--- a/Model/Import/Category.php
+++ b/Model/Import/Category.php
@@ -376,7 +376,10 @@ class Category extends AbstractEntity
         if (empty($this->categories)) {
             $stores = $this->storeManager->getStores();
             $searchStores = [\Magento\Store\Model\Store::DEFAULT_STORE_ID];
-            $this->nameToId['admin'] = \Magento\Store\Model\Store::DEFAULT_STORE_ID;
+            // BEGIN PATCH
+            //$this->nameToId['admin'] = \Magento\Store\Model\Store::DEFAULT_STORE_ID;
+            $this->nameToId['default'] = \Magento\Store\Model\Store::DEFAULT_STORE_ID;
+            // END PATCH
             foreach ($stores as $store) {
                 $this->nameToId[$store->getCode()] = $store->getId();
                 $searchStores[] = $store->getId();
@@ -603,10 +606,32 @@ class Category extends AbstractEntity
             $up = 0;
             $this->categoriesCache = [];
             $bunch = $this->prepareImagesFromSource($bunch);
+
+            // BEGIN PATCH
+            $categoryIds = array_map(function ($item) {
+                return $item['entity_id'];
+            }, $bunch);
+
+            $tableName = $this->_connection->getTableName('catalog_category_entity');
+
+            $select = $this->_connection->select()
+                ->from($tableName, ['entity_id'])
+                ->where('entity_id IN(?)', $categoryIds);
+
+            $existingCategoryIds = $this->_connection->fetchCol($select);
+            // END PATCH
+
             foreach ($bunch as $rowNum => $rowData) {
                 if ($rowData['name'] == 'Root Catalog') {
                     continue;
                 }
+
+                // BEGIN PATCH
+                if (!isset($rowData['entity_id']) || !in_array($rowData['entity_id'], $existingCategoryIds)) {
+                    continue;
+                }
+                // END PATCH
+
                 $this->_processedRowsCount++;
                 if (!empty($this->_parameters['use_only_fields_from_mapping'])) {
                     $rowData = $this->useOnlyFieldsFromMapping($rowData);
@@ -949,6 +974,11 @@ class Category extends AbstractEntity
         try {
             foreach (\array_keys($this->attrData) as $attrCode) {
                 if (!isset($rowData[$attrCode])) {
+                    // BEGIN PATCH
+                    if (is_object($category->getData($attrCode))) {
+                        continue;
+                    }
+                    // END PATCH
                     if ($category->getData($attrCode) == $defaultCategory->getData($attrCode)) {
                         if ($rowData['store_id'] != 0) {
                             $category->setData($attrCode, null);
@@ -1101,7 +1131,10 @@ class Category extends AbstractEntity
                             );
                     }
                 }
-                $lastPathCategoryId = array_last($categoryEntityIds);
+                // BEGIN PATCH
+                //$lastPathCategoryId = array_last($categoryEntityIds);
+                $lastPathCategoryId = end($categoryEntityIds);
+                // END PATCH
                 if ($lastPathCategoryId && $entityId) {
                     if ((int)$lastPathCategoryId !== $entityId) {
                         $this->getErrorAggregator()
@@ -1458,7 +1491,10 @@ class Category extends AbstractEntity
         if (!empty($rowData[self::COL_URL])) {
             $rowCategoryUrlKey = $rowData[self::COL_URL];
         } elseif (empty($rowData[self::COL_URL]) && !empty($rowData[self::COL_URL_PATH])) {
-            $rowCategoryUrlKey = array_last(explode(self::DELIMITER_CATEGORY, $rowData[self::COL_URL_PATH]));
+            // BEGIN PATCH
+            //$rowCategoryUrlKey = array_last(explode(self::DELIMITER_CATEGORY, $rowData[self::COL_URL_PATH]));
+            $rowCategoryUrlKey = end(explode(self::DELIMITER_CATEGORY, $rowData[self::COL_URL_PATH]));
+            // END PATCH
         }
         if (!empty($rowData[self::COL_PATH])) {
             $parentCategories = explode(self::DELIMITER_CATEGORY, $rowData[self::COL_PATH]);
@@ -1468,7 +1504,10 @@ class Category extends AbstractEntity
         }
         if (isset($rowData[self::COL_NAME])) {
             $parentCategoryNames = explode(self::DELIMITER_CATEGORY, $rowData[self::COL_NAME]);
-            $rowCategoryName = array_last($parentCategoryNames);
+            // BEGIN PATCH
+            //$rowCategoryName = array_last($parentCategoryNames);
+            $rowCategoryName = end($parentCategoryNames);
+            // END PATCH
             if (count($parentCategoryNames) > 2) {
                 $rowRootParentCategoryName = $parentCategoryNames[1];
             } elseif (count($parentCategoryNames) == 2) {
